Xabe.AutoUpdater/FtpProvider.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using Octokit;
+using FileMode = System.IO.FileMode;
+
+namespace Xabe.AutoUpdater
+{
+    /// <summary>
+    ///     Provider for FtpReleases
+    /// </summary>
+    public class FtpProvider : IReleaseProvider
+    {
+        private readonly string _ftpHosts;
+        private readonly string _ftpLogin;
+        private readonly string _ftpPassword;
+        private readonly string _ftpFolder;
+        private readonly bool _ftpPassiveMode;
+
+        /// <summary>
+        ///     Create instance of FtpProvider. 
+        /// </summary>
+        /// <param name="pHosts">FTP HOST.</param>
+        /// <param name="pLogin">FTP Account Login</param>
+        /// <param name="pPassword">FTP Account Password</param>
+        /// <param name="pPassiveMode">FTP Passive Mode</param>
+        /// <param name="pFolder">FTP Release Folder (optional)</param>
+        public FtpProvider(string pHosts, string pLogin, string pPassword, bool pPassiveMode, string pFolder = null)
+        {
+            _ftpHosts = pHosts;
+            _ftpLogin = pLogin;
+            _ftpPassiveMode = pPassiveMode;
+            _ftpPassword = pPassword;
+            _ftpFolder = string.IsNullOrEmpty(pFolder) ? "/" : pFolder;
+        }
+
+        /// <inheritdoc />
+        public async Task<List<string>> DownloadCurrentVersion()
+        {
+            return await Task.Run(async () =>
```

Comment:
Task.Run is unncessary.
If you remove it this will run the same without wasting cpu for creating new task.

Reply:
ok